### ISSUE TITLE
Very specific issue with Pos X and Target Bind making player push act oddly.

### ISSUE THREAD

**Issue by VinVulpis** [ikemen-engine/Ikemen-GO#1695]
### Describe the bug

So this is a very specific issue I've noticed with 2 characters made by The_None (Dee Bee Caw and Shaq) since the 0.99 update. It was new and seemed completely random, never thought much of it until recently. I decided to do some testing, and this is what I found:

It seems to very specifically happen when characters get target-bound with specific moves which causes a character walking INTO the target-bound character to teleport right through them after a brief push. It SEEMED random, my testing has found that it's entirely dependent on the POS X of the BOUND character.

If the pos-frozen character is on the left side of the screen or stage, the other character will teleport when moving left into them. Inversely the same happens on the right side of the screen or stage when moving right into the frozen character. 

**Video example:**
https://github.com/ikemen-engine/Ikemen-GO/assets/66293146/35a90bd0-c6cf-42c5-8254-506bb11a094d

This is not an issue specific to my own characters. I've tested this against characters created by others, as well as having other characters caught in the trap. Dee Bee Caw will teleport right through them just like in the video under the same exact circumstances. 

I was able to easily reproduce this issue myself with one of my own characters:
https://github.com/ikemen-engine/Ikemen-GO/assets/66293146/cce27f47-23b8-4983-bb4f-e3defdb5cc83
I made Aurumo's freeze projectile use a target bind and hang around instead of just a long pausetime and it happens. So it seems easy to reproduce with anything.

### To Reproduce

To Recreate this, you can get Dee Bee Caw by The_None/Most_Mysterious
https://www.justnopoint.com/lbends/index.php?location=chars&page=2 (4th character from the left on this page)

Have him use his Bear Trap special, (D,D, x)
Have either himself or opponent get caught in trap. 
Have other character try to walk against the trapped character at different positions. 

**Alternatively**, just make any helper cause a target-bind and walk into the bound character at  different stage/screen positions.

### Expected behavior

Expected behavior should not be able to walk through the trapped character at all.

### Engine Version (or source code date)

0.99.0

### Operating system

Windows

### Extra context or search terms

target_bind pos_x

**Comment by potsmugen** [comment]
This also happens in Mugen 1.1. The velocity threshold to trigger it is different, but not so different that I'd be worried.

In the current build you can avoid this issue with the "immovable" AssertSpecial flag.

**Comment by VinVulpis** [comment]
> This also happens in Mugen 1.1. The velocity threshold to trigger it is different, but not so different that I'd be worried.
> 
> In the current build you can avoid this issue with the "immovable" AssertSpecial flag.

I never noticed this back when I was using Mugen. Fun to know. I'm assuming it has soemthing to do with the camera zooming being added.

I DID HOWEVER  just realize that **"IkemenVersion"** was missing from Aurumo, the character of mine I tested the concept with. I added that in, and the teleport through no longer happens. I've since droped mugen in favor of goin IkemenGo exclusive, so simply adding in "IkemenVersion=0.99" fixes it.  When I saw it flagged as a "compatibility" issue, it gave me the idea to look and see if I added that.

Video with "Ikemenversion=0.99" added;
https://github.com/ikemen-engine/Ikemen-GO/assets/66293146/b643919a-828d-4d9c-a4bb-26982fb65a95

So I'll probs just edit other characters made for Mugen by adding that into the DEF as an lazy fix and hope it don't do anything else weird to them.
